sim/core/apl_actions_misc.go

```diff
@@ -93,6 +93,36 @@ func (action *APLActionActivateAura) String() string {
 	return fmt.Sprintf("Activate Aura(%s)", action.aura.ActionID)
 }
 
+type APLActionSetAuraStacks struct {
+	defaultAPLActionImpl
+	aura   *Aura
+	stacks int32
+}
+
+func (rot *APLRotation) newActionSetAuraStacks(config *proto.APLActionSetAuraStacks) APLActionImpl {
+	aura := rot.GetAPLAura(rot.GetSourceUnit(&proto.UnitReference{Type: proto.UnitReference_Self}), config.AuraId)
+	if aura.Get() == nil || aura.Get().MaxStacks == 0 {
+		return nil
+	}
+	return &APLActionSetAuraStacks{
+		aura:   aura.Get(),
+		stacks: int32(min(config.Stacks, aura.Get().MaxStacks)),
+	}
+}
+func (action *APLActionSetAuraStacks) IsReady(sim *Simulation) bool {
+	return true
+}
+func (action *APLActionSetAuraStacks) Execute(sim *Simulation) {
+	if sim.Log != nil {
+		action.aura.Unit.Log(sim, "Setting stacks on aura %s to %d", action.aura.ActionID, action.stacks)
+	}
+	action.aura.Activate(sim)
+	action.aura.SetStacks(sim, action.stacks)
+}
+func (action *APLActionSetAuraStacks) String() string {
+	return fmt.Sprintf("Set Aura Stacks(%s, %d)", action.aura.ActionID, action.stacks)
+}
```

Comment:
Same APL action in the SoD repo in case there's any differences
https://github.com/wowsims/sod/blob/29a0df85921fba14289e6cbeeb2aa3a9887c936d/sim/core/apl_actions_misc.go#L101-L142

Reply:
> Same APL action in the SoD repo in case there's any differences https://github.com/wowsims/sod/blob/29a0df85921fba14289e6cbeeb2aa3a9887c936d/sim/core/apl_actions_misc.go#L101-L142

Pretty much copied the naming and everything from the sod repo and force pushed :)